.github/workflows/main.yaml

```diff
@@ -34,6 +36,7 @@ jobs:
 
       - name: Deploy
         uses: JamesIves/github-pages-deploy-action@4.1.5
```

Comment:
Can you also update this to the latest release (4.6.8), because this version is still running on an deprecated node.js version and always warns:
> The following actions uses node12 which is deprecated and will be forced to run on node16: JamesIves/github-pages-deploy-action@4.1.5. For more info: https://github.blog/changelog/2023-06-13-github-actions-all-actions-will-run-on-node16-instead-of-node12-by-default/
>
> The following actions use a deprecated Node.js version and will be forced to run on node20: JamesIves/github-pages-deploy-action@4.1.5. For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/
>
> The `set-output` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Maybe this should just use the official https://github.com/actions/deploy-pages